MicrosoftDX/nether
# Issues/admin web ui

### Issue: #131 


haven't run CodeFormatter as this is work-in-progress

### Description:
Work-in-progress angular 2 admin UI. 

### Test:
As of now launching it involves a few manual steps which should be removed by the time it's ready.

1. Make sure node.js is installed (required to pull angular 2 packages).
2. Make sure npm, bower and gulp are installed in global namespace:
    - `npm install -g npm@latest`
    - `npm install -g bower`
    - `npm install -g gulp`
3. Navigate to Nether.Web directory
    - `npm install` to install angular dependencies
    - `bower install` to install bower dependencies
    - `gulp npmtolib` runs a task to copy a requires subset of angular 2 to wwwroot/lib
4. Run the web server `rundev.ps1`
5. In the browser navigate to http://localhost:5000/features/adminwebui/login and login with either `devuser:devuser` or `devadmin:devadmin`. Those have player and admin roles respectively so parts of the website will or won't work depending on permissions (there are no checks in UI now).
